addons/resource/models/resource.py

```diff
@@ -798,37 +727,35 @@ def compute_working_calendar(self, cr, uid, calendar_id=False, context=None):
         return wktime_cal
 
 
-class resource_calendar_leaves(osv.osv):
+class ResourceCalendarLeaves(models.Model):
     _name = "resource.calendar.leaves"
     _description = "Leave Detail"
-    _columns = {
-        'name' : fields.char("Name"),
-        'company_id' : fields.related('calendar_id','company_id',type='many2one',relation='res.company',string="Company", store=True, readonly=True),
-        'calendar_id' : fields.many2one("resource.calendar", "Working Time"),
-        'date_from' : fields.datetime('Start Date', required=True),
-        'date_to' : fields.datetime('End Date', required=True),
-        'resource_id' : fields.many2one("resource.resource", "Resource", help="If empty, this is a generic holiday for the company. If a resource is set, the holiday/leave is only for this resource"),
-    }
-
-    def check_dates(self, cr, uid, ids, context=None):
-        for leave in self.browse(cr, uid, ids, context=context):
-            if leave.date_from and leave.date_to and leave.date_from > leave.date_to:
-                return False
-        return True
 
-    _constraints = [
-        (check_dates, 'Error! leave start-date must be lower then leave end-date.', ['date_from', 'date_to'])
-    ]
+    name = fields.Char()
+    company_id = fields.Many2one(string="Company", related='calendar_id.company_id', store=True, readonly=True)
+    calendar_id = fields.Many2one("resource.calendar", string="Working Time")
+    date_from = fields.Datetime(string='Start Date', required=True)
+    date_to = fields.Datetime(string='End Date', required=True)
+    resource_id = fields.Many2one("resource.resource", string="Resource",
+                                  help="If empty, this is a generic holiday for the company. If a resource is set, the holiday/leave is only for this resource")
+
+    @api.multi
+    @api.constrains('date_from', 'date_to')
+    def check_dates(self):
+        for leave in self:
+            if leave.date_from and leave.date_to and leave.date_from > leave.date_to:
```

Comment:
date_from and date_to is required field. so no need to check "if leave.date_from and leave.date_to .